### DISCUSSION TITLE
HowTo: Use an NFS drive for "media" in Docker

### DISCUSSION THREAD

**Discussion by michaelsidenius** [automatic-ripping-machine/automatic-ripping-machine#1218] [Show and tell]
I have finally succeeded in using my NFS exported drive on another machine for the media storage while ripping (on Docker). Here is what worked.

- Use same user UID, GID on NFS server as the user running and starting ARM on your ripping machine.
- On the NFS server, ensure you have the right permissions with inheritance:
  - The NFS "Automatic-Ripping-Machine" folder must be set up with:
  - `chmod ugo+rwX Automatic-Ripping-Machine`
  - `chmod ugo+s   Automatic-Ripping-Machine`
  - `mkdir Automatic-Ripping-Machine/completed`
  - `mkdir Automatic-Ripping-Machine/movies`
  - `mkdir Automatic-Ripping-Machine/raw`
  - `mkdir Automatic-Ripping-Machine/transcode`
  - `chmod ugo+rwX Automatic-Ripping-Machine/*`
  - `chmod ugo+s   Automatic-Ripping-Machine/*`
- Create a local NFS docker volume for the container:
```
docker volume create \
  --driver local \
  --opt type=nfs \
  --opt o=addr=<NFSserver-name-or-IP>,rw \
  --opt device=":/<NFSserver-shared-folder-name>/Automatic-Ripping-Machine" \
  NFSvolume
docker volume inspect NFSvolume
```
- Start the Docker container like before, but map "/home/arm/media" to your newly created Docker NFS volume:
```
docker run -d \
    -p "8080:8080" \
    ....
    -v "NFSvolume:/home/arm/media" \
    ....
    automaticrippingmachine/automatic-ripping-machine:latest

```

**Comment by microtechno9000** [comment]
Nice

Could you raise a PR and add that into the ARM wiki? Other users would find that usefull.

Edit. Guide is here [Contribute-Wiki](https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/Contribute-Wiki)
